### PR TITLE
19034 rebase legal name branch

### PR DESCRIPTION
**Pull request by JazzarKarim**
*Issue #:* /bcgov/entity#19034

*Description of changes:*
Similar to https://github.com/bcgov/business-create-ui/pull/637

Files to look it:
- AssociationType.vue 
- ChangeBusinessType.vue

Other than that, I think it's OK. Small conflicts that I'm pretty sure are OK after I resolved them.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-edit-dev--pr-556-bnbysccb.web.app

File: src/components/common/YourCompany/ChangeBusinessType.vue [hunk 1]
```diff
@@ -390,7 +390,8 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
       return true
     }
     // Named ULC to BC Limited require a name request.
-    if (this.getOriginalLegalType === CorpTypeCd.BC_ULC_COMPANY && this.isBcLimited) {
+    if (this.getOriginalLegalType === CorpTypeCd.BC_ULC_COMPANY &&
```
Comment (severinbeauvais) [inline src/components/common/YourCompany/ChangeBusinessType.vue:393]: The old and new code is different from what's in main branch right now. (The line numbers are also the same -- 2 more lines of code in this branch.) I'd like to understand why.

https://github.com/bcgov/business-edit-ui/blob/201cd7691b1b8a0b95e8288cf0f982ea99b91f93/src/components/common/YourCompany/ChangeBusinessType.vue#L391

Comment (JazzarKarim) [reply src/components/common/YourCompany/ChangeBusinessType.vue:393]: This was also a part of your PR Sev: https://github.com/bcgov/business-edit-ui/commit/47b18fc6c851757daef4fb4b61470a07e761e06b

This is what is looks like in the feature branch:
https://github.com/bcgov/business-edit-ui/blob/2b4793d57065fbd25ac72e519a06a128550acb42/src/components/common/YourCompany/ChangeBusinessType.vue#L393

So, I made a manual resolution here that basically combines both.

Comment (severinbeauvais) [reply src/components/common/YourCompany/ChangeBusinessType.vue:393]: OK. Thanks for your patience with me! Resolving.

**Review by severinbeauvais — APPROVED** [review]

File: src/components/common/YourCompany/ChangeBusinessType.vue (hunk 1 above)
Comment (JazzarKarim) [reply src/components/common/YourCompany/ChangeBusinessType.vue:393]: Thank you for taking the time to have a close look Sev! ❤️

**Review by leodube-aot — APPROVED** [review]
LGTM

**Review by argush3 — APPROVED** [review]

**Pull request merged**